src/web3-utils.js

```diff
@@ -147,10 +157,165 @@ export function formatUnits(
   if (typeof truncateToDecimalPlace === 'number') {
     const [whole = '', dec = ''] = valueBeforeCommas.split('.')
     if (dec) {
-      const truncatedDec = dec.slice(0, truncateToDecimalPlace).replace(/0*$/, '')
+      const truncatedDec = dec
+        .slice(0, truncateToDecimalPlace)
+        .replace(/0*$/, '')
       valueBeforeCommas = truncatedDec ? `${whole}.${truncatedDec}` : whole
     }
   }
 
   return commas ? EthersUtils.commify(valueBeforeCommas) : valueBeforeCommas
 }
+
+export function useTokenReserves(tokenSymbol) {
+  const [tokenReserves, setTokenReserves] = useState(null)
+  const [loading, setLoading] = useState(false)
+  useEffect(() => {
+    let cancelled = false
+    let retryTimer
+    async function fetchTokenReserves() {
+      try {
+        setLoading(true)
+        const [tokenAddress] = getKnownContract(`TOKEN_${tokenSymbol}`)
+        if (!tokenAddress) {
+          setLoading(false)
+          throw new Error(`Unsupported token symbol: ${tokenSymbol}`)
+        }
+
+        const tokenData = await getTokenReserves(
+          tokenAddress,
+          Number(env('CHAIN_ID'))
+        )
+
+        if (!tokenData) {
+          setLoading(false)
+          throw new Error('Could not fetch reserves')
+        }
+        if (!cancelled) {
+          setTokenReserves(tokenData)
+          setLoading(false)
+        }
+      } catch (e) {
+        setLoading(false)
+        retryTimer = setTimeout(fetchTokenReserves, UNISWAP_MARKET_RETRY_EVERY)
+      }
+
+      return () => {
+        cancelled = true
+        clearTimeout(retryTimer)
+      }
+    }
+
+    fetchTokenReserves()
+  }, [tokenSymbol])
+
+  return [tokenReserves, loading]
+}
+
+// Wraps uniswap’s getMarketDetails() to only require
+// the other token in the pair, qualified using its symbol.
+async function getAnjMarketDetails(tokenSymbol) {
+  const [tokenData, anjData] = await Promise.all(
+    [tokenSymbol, 'ANJ'].map(symbol => {
+      // In the case of ETH, undefined should be passed as the reserves data.
+      if (symbol === 'ETH') {
+        return undefined
+      }
+
+      const [tokenAddress] = getKnownContract(`TOKEN_${symbol}`)
+      if (!tokenAddress) {
+        throw new Error(`Unsupported token symbol: ${symbol}`)
+      }
+      const tokenData = getTokenReserves(tokenAddress, Number(env('CHAIN_ID')))
+
+      if (!tokenData) {
+        throw new Error('Could not fetch reserves')
+      }
+      return tokenData
```

Comment:
Should we create a new function doing this, and use it from both here and `useTokenReserves()`?

Reply:
Actually, here and `useTokenReserves()` have different behaviors, due to this function returning `undefined` on `ETH`.

Reply:
Yes, I think it would make sense to return `undefined` in both cases? Even though we only use `useTokenReserves()` with ANJ.